Shareable Search on dashboard

SHyx0rmZ opened the pull request:
concourse/concourse#2265

SHyx0rmZ committed:
Parse query string for "search" and use it to initialize Dashboard search

SHyx0rmZ committed:
Resolve Maybe for search query as soon as possible

SHyx0rmZ committed:
QueryString.one expects a parser

SHyx0rmZ committed:
Modify URL to reflect dashboard search query

vito commented:
@mhuangpivotal Could you try this out sometime and merge if everything looks good?

SHyx0rmZ commented:
Hi @mhuangpivotal, thanks for catching that!

I sure can, might take me until Friday, though.

mhuangpivotal commented:
@SHyx0rmZ sounds good! Thanks.

SHyx0rmZ committed:
Empty search now changes URL to /? instead of /

Pushing an empty URL causes the history to not get modified. If we
really wanted to clean the URL of any trace of the search, we could
apparently do so by pushing location.pathname. I'm not sure this is
worth the additional work, so let's just render an empty QueryString.

SHyx0rmZ committed:
Add acceptance tests for URLs including dashboard searches

SHyx0rmZ commented:
I've tried to fix the edge case and add the acceptance tests.

I can, however, not run the tests to confirm they work, nor re-generate `bindata/bindata.go`, as the updated file doesn't look correct (JavaScript files generated by Elm shrink and `main.css` is only 15 bytes after the update).

SHyx0rmZ commented:
I'll try and fix more of my build environment over the next few days.

vito commented:
Could you let me know what the problem was once things are working? We
really need to iron out these dev workflow wrinkles.

On Fri, Jun 15, 2018, 4:09 AM Patrick Pokatilo <notifications@github.com>
wrote:

> I'll try and fix more of my build environment over the next few days.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/concourse/web/pull/4#issuecomment-397547113>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAHWAPj6l9Pip8J6T_j5GJhKXlaiIhBks5t82vPgaJpZM4UhMTx>
> .
>

mhuangpivotal commented:
@vito Does it make sense to include a section in [`CONTRIBUTING.md`](https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md) on developing for web?

@SHyx0rmZ FYI you can generate the `bindata/bindata.go` by running `make clean all`. Running acceptance tests should only require `ruby` and the following commands:

```
cd acceptance
gem install bundler
bundle
bundle exec rspec spec/<some_spec.rb>
```

vito commented:
Yeah, I've been wanting to do that and make the repo layout more like a
conventional JavaScript project (potentially retiring the Ruby suite so
it's all one dev stack). For now we should at least document the workflow
and then figure out how to improve it, cause it's gotten pretty confusing
imo.

On Fri, Jun 15, 2018, 9:46 AM Mark Huang <notifications@github.com> wrote:

> @vito <https://github.com/vito> Does it make sense to include a section
> in CONTRIBUTING.md
> <https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md> on
> developing for web?
>
> @SHyx0rmZ <https://github.com/SHyx0rmZ> FYI you can generate the
> bindata/bindata.go by running make clean all. Running acceptance tests
> should only require ruby and the following commands:
>
> cd acceptance
> gem install bundler
> bundle
> bundle exec rspec spec/<some_spec.rb>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/concourse/web/pull/4#issuecomment-397624803>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAHWGc4aTnNThpL8mnKJDPtV49lL7T9ks5t87qjgaJpZM4UhMTx>
> .
>

SHyx0rmZ commented:
OK, I got it to a somewhat working state. My problem was mostly with guessing which packages to install via `npm`. I also managed to finally get `dev/start` working. I worked with Ruby before, so that stuff posed to problem for me.

SHyx0rmZ committed:
Update bindata/bindata.go

SHyx0rmZ commented:
@mhuangpivotal I updated the `bindata/bindata.go` now.

SHyx0rmZ committed:
Merge remote-tracking branch 'origin/master' into links-including-dashboard-search

mhuangpivotal commented:
@SHyx0rmZ Thanks for the update!

Sorry for the delay. You are one of the first to submit a web PR, and we are in the process of setting up CI integration (so the CI check below will not be accurate until concourse/concourse#2305 is complete).

It seems like a few other tests are breaking from this change. Specifically, the tests related to searching in the dashboard view. You can run the tests and see the failures by running:

```
cd acceptance/
bundle exec rspec spec/dashboard_search_spec.rb
```

Would you be able to take a look?

SHyx0rmZ commented:
I'll need to figure out how to make *any* test pass first, but I'll definitely have a look. Thanks for being patient with me!

SHyx0rmZ commented:
While I have managed to get *some* tests to pass, I'm still getting random errors all over the place, so it's probably going to take me some more time. Sorry :(

SHyx0rmZ committed:
Merge branch 'master' into links-including-dashboard-search

SHyx0rmZ committed:
Use backspaces to reset search input field in tests

SHyx0rmZ committed:
Update bindata/bindata.go

SHyx0rmZ committed:
Merge branch 'master' into links-including-dashboard-search

SHyx0rmZ commented:
I finally figured out how to properly run the tests. I forgot to set `ATC_URL` when invoking `rspec`. Will now work on fixing the remaining problems.

mhuangpivotal commented:
@SHyx0rmZ Thanks for continuing to work on this PR!

 Right now the CI pipeline is only triggered when the PR is approved, so let us know when you are ready for us to review it.

SHyx0rmZ committed:
Remove unnecessary duplicate import

SHyx0rmZ committed:
Update bindata/bindata.go

SHyx0rmZ committed:
Filter pipelines after they have been fetched

SHyx0rmZ committed:
Merge branch 'master' into links-including-dashboard-search

SHyx0rmZ commented:
@mhuangpivotal I'd be happy if you could review it. I think I managed to solve most of the problems.

SHyx0rmZ committed:
Replace '+' with ' ' in search query

SHyx0rmZ committed:
Reuse DashboardModel if route changes

SHyx0rmZ committed:
Remove unnecessary clean-up code

vito commented:
@mhuangpivotal I'm gonna remove the restriction that PRs need approval before the job can run. It should be safe to do that now.

mhuangpivotal commented:
We broke the tests on master, hence the CI failure. We are just going to merge it now to avoid future conflicts.
